processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -294,6 +295,7 @@ public void writeIndexes(List<IntBuffer> segmentRowNumConversions, Closer closer
     try (
         Closeable toCloseEncodedValueWriter = encodedValueWriter;
         Closeable toCloseBitmapWriter = bitmapWriter;
+        Closeable toCloseDictionaryMergeIterator = dictionaryMergeIterator;
```

Comment:
The dictIdSeeker and dimConversions in this block use the IntBuffer views created from the ByteBuffers in the iterator, it's needed until after mergeBitmaps()